### PR TITLE
Fix url deps and add eslint rules to not allow core nodejs modules on lts

### PR DESCRIPTION
**Pull request by RishhiB**
Affects the following packages:

moving this fix to LTS

packages\drivers\local-driver
packages\drivers\routerlicious-urlResolver
packages\test\test-end-to-end-tests
packages\tools\fetch-tool - removed unnecessary url import since it's a browser api now
packages\loader\container-loader

### CODE REVIEW

File: packages/test/test-end-to-end-tests/.eslintrc.js [hunk 1]
```diff
@@ -4,14 +4,28 @@
  */
 
 module.exports = {
-    "extends": [
-        require.resolve("@fluidframework/eslint-config-fluid")
-    ],
-    "rules": {
+    extends: [require.resolve("@fluidframework/eslint-config-fluid")],
+    rules: {
```
Comment (tylerbutler) [inline packages/test/test-end-to-end-tests/.eslintrc.js:8]: nit: try to use the order:

extends
parserOptions
rules
overrides

File: packages/test/test-end-to-end-tests/.eslintrc.js [hunk 2]
```diff
@@ -4,14 +4,28 @@
...
-}
+    parserOptions: {
+        project: ["./src/test/tsconfig.json"],
+    },
+    overrides: [
+        {
+            // Rules only for test files
+            files: ["*.spec.ts", "src/test/**"],
+            rules: {
+                // This library is used in the browser, so we don't want dependencies on most node libraries.
```
Comment (tylerbutler) [inline packages/test/test-end-to-end-tests/.eslintrc.js:23]: Nit: maybe "Test files are run in node only so additional node libraries can be used.

File: packages/tools/fetch-tool/src/fluidFetchArgs.ts [hunk 3]
```diff
@@ -4,7 +4,6 @@
  */
 
 import fs from "fs";
-import { URL } from "url";
```
Comment (tylerbutler) [inline packages/tools/fetch-tool/src/fluidFetchArgs.ts:7]: Was this not used?

File: packages/tools/fetch-tool/src/fluidFetchInit.ts [hunk 4]
```diff
@@ -3,7 +3,6 @@
  * Licensed under the MIT License.
  */
 
-import { URL } from "url";
```
Comment (tylerbutler) [inline packages/tools/fetch-tool/src/fluidFetchInit.ts:6]: Same here.

Comment (RishhiB) [reply packages/tools/fetch-tool/src/fluidFetchInit.ts]: Added both imports again

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**